### PR TITLE
avoid static param table size

### PR DESCRIPTION
**Pull request by mdavidsaver**
As described in #35 remove static limit on parameter size.  Also create asynPortDriverTest to exercise parameter creation and change flag code effected by this change.  Remove unused (and un-buildable) ParamListTest.cpp and ParamValTest.cpp.

Also some bug fixes found along the way.  An incorrect error check in ```asynPortClient::asynPortClient```.  Freeing of otherwise leaked interruptNodes in all *Base except asynOctetBase.  Finally, make ```~asynInt32Client``` call cancelInterruptUser if necessary.  Other *Client classes will need this as well, but I'm out of time for today.

### CODE REVIEW

File: asyn/asynPortDriver/asynPortDriver.cpp [hunk 1]
```diff
@@ -77,66 +80,64 @@ class paramList {
...
     if (this->findParam(name, index) == asynSuccess) return asynParamAlreadyExists;
-    *index = this->nextParam++;
-    if (*index < 0 || *index >= this->nVals) return asynParamBadIndex;
-    delete this->vals[*index];
-    this->vals[*index] = new paramVal(name, type);
+
+    std::auto_ptr<paramVal> param(new paramVal(name, type));
+
+    if (this->vals.size() == maxParams) {
+        asynPrint(pasynPortDriver->pasynUserSelf, ASYN_TRACE_ERROR,
```
Comment (ulrikpedersen) [inline asyn/asynPortDriver/asynPortDriver.cpp]: Use ASYN_TRACE_WARNING instead of ERROR?

File: asyn/asynPortDriver/asynPortDriver.cpp [hunk 2]
```diff
@@ -146,8 +147,11 @@ asynStatus paramList::createParam(const char *name, asynParamType type, int *ind
   * \return Returns asynParamNotFound if name is not found in the parameter list. */
 asynStatus paramList::findParam(const char *name, int *index)
 {
-    for (*index=0; *index<this->nVals; (*index)++) {
-        if (this->vals[*index]->nameEquals(name)) return asynSuccess;
+    for (size_t i=0; i<this->vals.size(); i++) {
```
Comment (ulrikpedersen) [inline asyn/asynPortDriver/asynPortDriver.cpp]: vals is a vector - why not use an iterator over vals?

File: asyn/asynPortDriver/asynPortDriver.cpp (hunk 1 above)
Comment (mdavidsaver) [inline asyn/asynPortDriver/asynPortDriver.cpp]: will do

File: asyn/asynPortDriver/asynPortDriver.cpp (hunk 2 above)
Comment (mdavidsaver) [inline asyn/asynPortDriver/asynPortDriver.cpp]: vector is the one container where this makes no practical difference.  It is a smaller change, and easier to validate, to keep the indexed access.

**Pull request merged**